lib/syncthing/utils.go

```diff
@@ -24,6 +24,29 @@ import (
 	"github.com/syncthing/syncthing/lib/tlsutil"
 )
 
+func EnsureDir(dir string, mode fs.FileMode) error {
+	fs := fs.NewFilesystem(fs.FilesystemTypeBasic, dir)
+	err := fs.MkdirAll(".", mode)
+	if err != nil {
+		return err
+	}
+
+	if fi, err := fs.Stat("."); err == nil {
+		// Apprently the stat may fail even though the mkdirall passed. If it
+		// does, we'll just assume things are in order and let other things
+		// fail (like loading or creating the config...).
```

Comment:
I know this isn't yours, but this sounds mysterious, and not in good way.

Reply:
Yeah, I just moved the code, didn't really understand this comment either.  Could we postpone examination and fixing to another PR?  I at least wouldn't know what circumstances to test and under what conditions such behavior may have been observed.  Might be worth keeping that commit separate so tracking that it was only moved is more obvious?